### PR TITLE
Fixes #30380 - enforce permissions on repository contr

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
This PR provides two ways of enforcing object level permissions:

1.  The simple way:  https://github.com/Katello/katello/pull/8881/commits/03855768b17060335c4182bb6524fd6893fa7a3a
This method would not be very DRY across multiple repositories but is very obvious.
2.  The more complex, but DRYier way https://github.com/Katello/katello/pull/8881/files
This has the advantage that the permissions are not repeated in the controller, and i like it quite a bit.  The only downside is that we are monkey patching the permission registry object, but its very simple.   The reason we need to associate a 'permission scope (i.e. Product.readable), with a controller, whereas foreman doesn't, is that we have instances of inherited permissions.  Product edit gives the ability to edit repositories.    So there isn't an easy way to map from one to the other without doing something like this.

**Comment by ianballou** [comment]
@jlsherrill how can I reproduce this?   I've been playing around with a user's permissions and I haven't been able to see or create and repositories that I shouldn't be able to.

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -438,16 +440,12 @@ def gpg_key_content
...
       find_organization_from_product if @organization.nil? && @product
     end
 
     def find_product_for_create
-      @product = Product.find(params[:product_id])
-    end
-
-    def find_repository
-      @repository = Repository.find(params[:id])
+      @product = Product.editable.find(params[:product_id])
```
[the remark this answers is not in the document]
Comment (jlsherrill) [reply app/controllers/katello/api/v2/repositories_controller.rb]: fixed!

[the remark this answers is not in the document]
Comment (jlsherrill) [reply app/controllers/katello/api/v2/repositories_controller.rb]: but i wasn't sure what you meant by 'I think find_authorized_katello_resource isn't being used along with the unauthorized method.'

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jlsherrill** [comment]
rebase against master had a mistake, fixed!

**Pull request merged**